File: backend/services/googleSpeech.js
const speech = require('@google-cloud/speech');
const path = require('path');

const client = new speech.SpeechClient({
  keyFilename: path.resolve(process.env.GOOGLE_APPLICATION_CREDENTIALS),
});

const transcribeAudio = async (gcsUri) => {
  const audio = { uri: gcsUri }; // Use gs:// URI
  const config = {
    encoding: 'LINEAR16', // Ensure this matches your file type
    languageCode: 'en-US',
  };
  const request = { audio, config };

  const [response] = await client.recognize(request);
  const transcription = response.results
    .map((result) => result.alternatives[0].transcript)
    .join('\n');

  return transcription;
};

module.exports = { transcribeAudio };

File: backend-python/services/google_speech.py
from google.cloud import speech
import os

def transcribe_audio(gcs_uri):
    client = speech.SpeechClient.from_service_account_json(os.getenv("GOOGLE_APPLICATION_CREDENTIALS"))

    audio = speech.RecognitionAudio(uri=gcs_uri)
    config = speech.RecognitionConfig(
        encoding=speech.RecognitionConfig.AudioEncoding.LINEAR16,
        language_code="en-US",
    )
    
    response = client.recognize(config=config, audio=audio)
    return " ".join(result.alternatives[0].transcript for result in response.results)
